setup.py

```
@@ -0,0 +1,176 @@
+    links = [path.realpath(x) for x in links]
+    link_args = ["-Wl,-rpath,{}".format(p) for p in links]
+
+    with open('compiler.txt','r') as f:
+        cc = f.read().strip()
+    environ["CC"] = cc
+
+    with open('libs.txt','r') as f:
+        libs = [x. strip() for x in f.readlines()]
+    libs = reduce(lambda x,y: x+y,[x.split(' ') for x in libs])
+    libs = [x.lstrip('-l') for x in libs]
+
+    with open('lib_sources.txt','r') as f:
+        lib_sources = f.read().strip().split(' ')
+except IOError:
+    raise IOError("There is missing metadata. Try running "
+                  +"'make meta' to generate it.")
+
+
+all_object_files = glob.glob(path.join(here,'*.o'))
+if len(all_object_files) == 0:
+    raise IOError("There are missing object files! Try running "
+                  +"'make' to build SimulationIO.")
+object_files = [src.replace('cpp','o') for src in lib_sources]
+
+swig_modules = ['H5','SimulationIO','RegionCalculus']
+swig_module_files = [path.join(here,"{}.py".format(m)) for m in swig_modules]
+swig_module_targets = [path.join(here,'pysimulationio',"{}.py".format(m))\
+                       for m in swig_modules]
+swig_wrap_files = (
+    [path.join(here,"{}_wrap.cpp".format(m)) for m in swig_modules]
+    +[path.join(here,"{}_wrap.cxx".format(m)) for m in swig_modules]
+    +[path.join(here,"{}_wrap.o".format(m)) for m in swig_modules])
+swig_opts = ['-c++','-Wall']
+
+# Overload the build and install commands so that swig behaves. See:
+# stackoverflow.com/questions/12491328/python-distutils-not-include-the-swig-generated-module
+# I would consider this a bug in setuptools.
+class CustomBuild(build):
+    def run(self):
+        self.run_command('build_ext')
+        # We copy the *.py files generated by swig by hand.
+        # TODO: find cleaner way to handle this
+        for i in range(len(swig_modules)):
+            if path.exists(swig_module_targets[i]):
+                os.remove(swig_module_targets[i])
```

The pythonic way of doing this: Unconditionally remove the file, and ignore errors if the file doesn't exist.